### PR TITLE
Feature/ready indicator

### LINKED ISSUE #77: Add player ready indicator
Add an indicator for showing how many players (of those still participating) are currently ready.

Opened from Discord by **Nibuja#6654**
Follow the conversation [here](https://discord.com/channels/501306949434867713/826466764606799882/841009073691426836)

### PR DESCRIPTION
**Pull request by pussyslain**
- all basic ui atm
- add chat cmd `-forceallplayerready` for testing

https://www.loom.com/share/ad59a2e77d53432682b374a1122b3394

Fixes #77

### CODE REVIEW

**Comment by Shushishtok** [comment]
@pussyslain Can you please add:
"Fixes https://github.com/arcadia-redux/reVolt/issues/77" to your description?

File: content/panorama/layout/custom_game/wave_counter.xml [hunk 1]
```diff
@@ -27,23 +27,29 @@
...
+                <Image id="PreparationPhase" class="WavePhaseIcon active" src="file://{images}/custom_game/preparation_icon.png" />
+            </Panel>
+            <Panel id="WaveTimer">
+                <Label id="WaveTimerLabel" text="0:00" />
+            </Panel>
+        </Panel>
         <Panel id="NextWave">
+            <Panel id="NextWaveReadyContainer">
+                <Image id="NextWaveReadyIcon" src="file://{images}/custom_game/check_icon.png" />
+                <Label id="NextWaveReadyLabel" text="Ready" />
```
Comment (Nibuja05) [inline content/panorama/layout/custom_game/wave_counter.xml:47]: Can you localize all these strings pls?
To do that change it here to something like `#ui_wave_ready` and add `ui_wave_ready` to the UI localization file

File: content/panorama/layout/custom_game/wave_counter.xml [hunk 2]
```diff
@@ -27,23 +27,29 @@
...
+                <Label id="NextWaveReadyCounter" text="" />
+                <Label id="NextWaveReadyCountdown" text="" />
+            </Panel>
             <Button id="NextWaveButton" onactivate="Ready();">
-                <Label id="NextWaveLabel" text="Start" />
+                <Label id="NextWaveLabel" text="Ready" />
```
Comment (Nibuja05) [inline content/panorama/layout/custom_game/wave_counter.xml:52]: localization here too

File: content/panorama/scripts/custom_game/wave_counter.ts [hunk 3]
```diff
@@ -85,9 +94,28 @@ function SetWaveTime(time: number) {
...
+	let readyCounterLabel = $("#NextWaveReadyCounter") as LabelPanel;
+	let readyCountdownLabel = $("#NextWaveReadyCountdown") as LabelPanel;
+
+	if (numPlayersReadyForNextWave === 0) {
+		readyCounterLabel.text = "";
+		readyCountdownLabel.text = "";
+	} else if (numPlayersReadyForNextWave >= (maxPlayers || 0)) {
+		// all ready
+		readyCounterLabel.text = "";
+		readyCountdownLabel.text = "Starting...";
```
Comment (Nibuja05) [inline content/panorama/scripts/custom_game/wave_counter.ts:107]: localize here too

File: content/panorama/styles/custom_game/wave_counter.css [hunk 4]
```diff
@@ -141,6 +141,64 @@
...
+	opacity: 0;
+}
+
+#NextWaveReadyLabel {
+	align: middle middle;
+	height: 40px;
+	width: 100px;
+	margin-bottom: 4px;
+	font-size: 30px;
+	font-family: Reaver SemiBold;
```
Comment (Nibuja05) [inline content/panorama/styles/custom_game/wave_counter.css:158]: I didn't know that before, but specifying the font like that won't be rendered correctly for chinese, russian etc.
I know that I missed that in a lot of placed previously, but could you pls change all occurances of it? TY!

Here is a little more information about what I'm talking about and how to fix it: [fonts](https://gist.github.com/Nibuja05/024867cf435e171798d21911ceceb8d5)

To add the `SemiBold` property, use this instead: `font-weight: 600;`

Sry for the inconveniance 😅

**Review by Nibuja05 — COMMENTED** [review]
Looking really good in general, especially for your first issue, that is really impressive!!
Some minor things here and there, so it will be easy to adjust.

Some additional notes:
- I really like adding some transitions for opacity and stuff here and there (even if its really minor) to make it a little more pleasant for the eyes ^^
Not really necessary here tho, as we have an UI look overhaul somewhere in the future anyways
- You added a new dev command in this PR (good addition for sure), but there is currently no button for it in the devUI. Since its something we probably won't use very often its ok (also there's not much space left), but just keep in mind that for bigger additions its very useful to add a new button there too.

**Comment by pussyslain** [comment]
closing this with respect to #136

**Pull request closed without merging**